SharpSploit/Execution/PE.cs

```diff
@@ -48,9 +84,16 @@ public bool Is32BitHeader
 
         /// The DOS header
         private IMAGE_DOS_HEADER dosHeader;
-        //Primary class for loading PE
-        [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
-        private delegate bool main(IntPtr arg1, uint arg2, IntPtr lparam);
+
+        // DllMain constants
+        public const UInt32 DLL_PROCESS_DETACH = 0;
+        public const UInt32 DLL_PROCESS_ATTACH = 1;
+        public const UInt32 DLL_THREAD_ATTACH = 2;
+        public const UInt32 DLL_THREAD_DETACH = 3;
+
+        // Primary class for loading PE
+        [UnmanagedFunctionPointer(CallingConvention.StdCall)]
+        public delegate bool DllMain(IntPtr hinstDLL, uint fdwReason, IntPtr lpvReserved);
 
         private static IntPtr codebase;
 
```

Comment:
I feel like this PE class now duplicates a lot of logic now present in SharpSploit.Execution.DynamicInvoke.Generic.cs. Would it be possible to merge/transition this logic over to the new code?

Reply:
We can take a look and see how much of it we can merge into the new code. We pretty much re-implemented all of it as part of building the manual mapping, so not sure if we need any of it any more.

Reply:
Will leave as-is for now until we fix the manual mapper for WOW64 processes.